keymanager API: bug fixes and inconsistencies (#14586)
prysmaticlabs/prysm: CHANGELOG.md

  cache impossible. This has been fixed with updates to rules_oci and bazel-lib.
- Fixed an issue where the length check between block body KZG commitments and the existing cache from the database was incompatible.
- Fix `--backfill-oldest-slot` handling - this flag was totally broken, the code would always backfill to the default slot [pr](https://github.com/prysmaticlabs/prysm/pull/14584)
- Fix keymanager API should return corrected error format for malformed tokens

> - Fix keymanager API should return empty instead of error on get keys API when using the unsupported keystore instead of failing with 500 error

Reviewer: Nit
```suggestion
- Fix keymanager API so that get keys returns an empty response instead of a 500 error when using an unsupported keystore.
```
